ambry-clustermap/src/main/java/com/github/ambry/clustermap/AmbryPartitionStateModel.java

```diff
@@ -194,14 +196,23 @@ public void reset() {
   boolean shouldTransition(Message message) {
     String resourceName = message.getResourceName();
     String partitionName = message.getPartitionName();
+    String resourceNameToCompare = clusterManager.getResourceForPartitionInLocalDc(partitionName);
+    if (resourceNameToCompare != null && !resourceNameToCompare.equals(resourceName)) {
```

Comment:
Looks like the resource name is an incremental integer. 
I guess the following code is trying to get the latest resource for this partition, right?
Do you mind adding some comment for it.